src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -173,6 +175,59 @@ private Result CreateBaselinedResult(BaselineState newBaselineState)
             return result;
         }
 
+        internal void SetFirstDetectionTime(Result targetResult)
```

Comment:
Please sanity test with a multitool match-results-forward case just to make sure this logic runs from end-to-end result matching also. I don't think this needs to be unit test necessarily - the unit tests and a one-off verification that everything is really hooked up is probably fine.

Reply:
Thanks for the prodding :-) I made a copy of a test file with two issues. In the copy I added one, removed one of the two originals, and left the third alone. `provenance.firstDetectionTimeUtc` is populated as expected.

---
In reply to: [372135643](https://github.com/microsoft/sarif-sdk/pull/1739#discussion_r372135643) [](ancestors = 372135643)